### PR TITLE
feat(animation): Implement `AnimationDirection`

### PR DESCRIPTION
**Pull request by SK83RJOSH**
This PR implements `PropertyAnimation.direction` as mentioned in #6260, but not `PropertyAnimation.fill-mode`. Slint's default fill mode could be considered `both`, and as such this PR limits the scope to implementing only that fill mode.

If desired, adding support for `AnimationFillMode` could be done, using the following enum:
```rust
enum AnimationFillMode {
    None = 0,
    Forwards = 1,
    Backwards = 2,
    Both = 3,
}
```

And using bit masks:
```rust
(fill_mode & AnimationFillMode::Forwards) != 0
(fill_mode & AnimationFillMode::Backwards) != 0
```

However, doing so would require fetching the animated property *prior* to it being set by the current state when `None`, `Forwards`, or `Backwards` are chosen. Which may prove tricky, and may not currently be possible. Thus my decision to skip the extra complexity for now and keep this PR as bare bones as possible.

Here are some videos demonstrating the feature:

https://github.com/user-attachments/assets/bbdb6082-c957-4036-94d3-5be79ab50cc8

https://github.com/user-attachments/assets/96f4ae62-8f5c-4d4f-bed5-8beace269923

https://github.com/user-attachments/assets/86d76a43-1f55-42d9-8c07-57c652e09513

### CODE REVIEW

**Comment by SK83RJOSH** [comment]
Alrighty, I've tested the lsp, added a test, and updated every bit of related code/documentation I could find.

The only remaining question I have, is whether or not any other code might depend on the old behavior? I noticed that very rarely, widgets revert to `to_value` when the layout is refreshed (by fiddling with other widgets/resizing the window). Not sure if it's simply an existing bug with `from`/`to` states, or something I have caused by breaking assumptions made elsewhere.

I would say it's a non-blocker for merging given the rarity, but I'd love to pinpoint the issue if possible. ❤️

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ All notable changes to this project are documented in this file.
  - The `mod` function was changed to always return a positive value (#6178)
  - Added `AccesssibleRole.list-item` and used it where relevant
  - Added `PointerEventButton.back` and `PointerEventButton.forward` (#6043)
+ - Added `PropertyAnimation.direction` property for controlling animation direction (#6261).
```
Comment (ogoffart) [inline CHANGELOG.md:25]: Please put Remove the line from the CHANGELOG file and put it in a commit message instead.

**Review by ogoffart — COMMENTED** [review]
Thanks a lot for your PR. 🤗

This is very complete with tests and docs, appreciated.

Please rebase to fix the conflict (remove the line in the changelog) and this is good to go.

A test in a .slint file would be extra nice (for example like tests/cases/properties/property_animation.slint) (doesn't need to have all languages, one is enough)

**Comment by SK83RJOSH** [comment]
Done! Added tests for all languages already in that file, though took a bit longer than I'd like due to rust compile times and the complexity of the test adapter. Time to see if the CI succeeds. 🙂

**Comment by ogoffart** [comment]
> took a bit longer than I'd like due to rust compile times

Yes, the test takes ages to compile; you can use the SLINT_TEST_FILTER env variable to only compile a single test to make it faster
https://github.com/slint-ui/slint/blob/master/docs/testing.md#rust-driver

**Review by ogoffart — APPROVED** [review]
Great!

File: internal/core/properties/properties_animations.rs [hunk 2]
```diff
@@ -71,18 +86,32 @@ impl<T: InterpolatedPropertyValue + Clone> PropertyValueAnimationData<T> {
...
+                        }
+                    };
                     let t = crate::animations::easing_curve(&self.details.easing, progress);
                     let val = self.from_value.interpolate(&self.to_value, t);
 
                     (val, false)
                 } else {
-                    self.state = AnimationState::Done;
+                    self.state =
+                        AnimationState::Done { interation_count: (current_iteration - 1).max(0) };
```
Comment (SK83RJOSH) [inline internal/core/properties/properties_animations.rs:104]: Whoops, just noticed these are unsigned, so this `max` is useless - I'm pretty sure it's never going to underflow anyway, given we bail out on `iteration_count == 0.`.

**Comment by ogoffart** [comment]
Thanks!

**Comment by tronical** [comment]
Awesome feature and implementation:)

**Pull request merged**